pandas/tests/test_categorical.py

```diff
@@ -57,6 +57,13 @@ def test_getitem_listlike(self):
         expected = c[np.array([100000]).astype(np.int64)].codes
         tm.assert_numpy_array_equal(result, expected)
 
+    def test_getname_category(self):
+        expected = 'A'
+        s = pd.Series([1, 2, 3], name='A').astype('category')
+        s = s.cat.set_categories([1, 2, 3])
+        result = s.astype('category').name
```

Comment:
This should be `result == s.cat.set_categories([1, 2, 3]).name` so that we're directly testing the behavior that caused the bug.